src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -142,6 +142,56 @@ private static TimeZoneInfo GetLocalTimeZoneCore()
             return Utc;
         }
 
+        private static TimeSpan GetCacheLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+                CachedData cachedData = s_cachedData;
+                return cachedData.Local.GetUtcOffset(dateTime, flags, cachedData);
+        }
+
+        private static TimeSpan? _localUtcOffset;
+        private static object _localUtcOffsetLock = new();
+        private static Thread? _loadAndroidTZData;
+        // Shortcut for TimeZoneInfo.Local.GetUtcOffset
+        // On Android, loading AndroidTZData while obtaining cachedData.Local is expensive for startup.
+        // We introduce a fast result for GetLocalUtcOffset that relies on the date time offset being
+        // passed into monovm_initialize(_preparsed) from Java in seconds as LOCAL_DATE_TIME_OFFSET.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // The fast path is initially used, and we start a background thread to get cachedData.Local
+        internal static TimeSpan GetLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+            if (_localUtcOffset != null) // The background thread finished, the cache is loaded.
+                return GetCacheLocalUtcOffset(dateTime, flags);
+
+            if (_localUtcOffset == null && _loadAndroidTZData == null) // The cache isn't loaded and no background thread has been created
+            {
+                lock (_localUtcOffsetLock)
+                {
+                    // GetLocalUtcOffset may be called multiple times before a cache is loaded and a background thread is running,
+                    // once the lock is available, check for a cache and background thread.
+                    if (_localUtcOffset != null)
+                        return GetCacheLocalUtcOffset(dateTime, flags);
+
+                    if (_loadAndroidTZData == null)
+                    {
+                        _loadAndroidTZData = new Thread(() => {
+                            _localUtcOffset = GetCacheLocalUtcOffset(dateTime, flags);
+                            Thread.Sleep(1000);
```

Comment:
I think @grendello was suggesting to move the sleep up one line here.

Reply:
I have the commit in a pending commit. I'll push it up, I was trying to bundle both this change and anything to do with parsed properties, but ill just separate.